src/parsers/power_models_data.jl

```diff
@@ -170,17 +189,31 @@ function read_bus!(sys::System, data::Dict; kwargs...)
     bus_number_to_bus = Dict{Int, ACBus}()
 
     bus_types = instances(ACBusTypes)
-    bus_data = sort!(collect(data["bus"]); by = x -> parse(Int, x[1]))
-
+    unique_bus_names = true
+    bus_data = SortedDict{Int, Any}()
+    # Bus name uniqueness is not enforced by PSSE. This loop avoids forcing the users to have to
+    # pass the bus formatter always for larger datasets.
+    bus_names = Set{String}()
+    for (k, b) in data["bus"]
+        # If buses aren't unique stop searching and growing the set
+        if unique_bus_names && haskey(b, "name")
+            if b["name"] ∈ bus_names
+                unique_bus_names = false
+            end
+            push!(bus_names, b["name"])
+        end
+        bus_data[k] = b
+    end
     if isempty(bus_data)
         @error "No bus data found" # TODO : need for a model without a bus
     end
 
-    _get_name = get(kwargs, :bus_name_formatter, _get_pm_dict_name)
+    default_bus_naming = x -> _get_pm_bus_name(x, unique_bus_names)
+
+    _get_name = get(kwargs, :bus_name_formatter, default_bus_naming)
     for (i, (d_key, d)) in enumerate(bus_data)
         # d id the data dict for each bus
         # d_key is bus key
-        d["name"] = get(d, "name", string(d["bus_i"]))
         bus_name = strip(_get_name(d))
         bus_number = Int(d["bus_i"])
 
```

Comment:
The error message on line 230 could be confusing. If this code and its assumptions are correct, the only way for the error to occur is if the user provided a custom bus_formatter function that did not produce unique names.

Reply:
I just updated the error message